vortexasdk/api/shared_types.py

```diff
@@ -1,90 +1,118 @@
 from abc import ABC
-from dataclasses import dataclass
+from pydantic import BaseModel
 from datetime import datetime
 from typing import List, Optional, Union
 
 from vortexasdk.api.id import ID
 
 IDsNames = Union[List[Union[ID, str]], str, ID]
 
-ISODate = str
+ISODate = datetime
```

Comment:
`ISODate` is just an alias for `datetime`? Worth considering given that we use them both in our types